.github/workflows/publish.yaml

```diff
@@ -7,7 +7,7 @@ on:
   pull_request:
     branches: [ main ]
   push:
-    tags: [ '[A-z]+-v[0-9]+.[0-9]+.[0-9]+' ]
+    tags: [ '[A-z0-9]+-v[0-9]+.[0-9]+.[0-9]+' ]
```

Comment:
Escape the `.` too. This regexp allows `foo-v0x1/2`.
I don't know how many `\`s are needed, _probably_ `\\.`, so maybe use `[.]` instead.

If the first part is a [Pub package name](https://dart.dev/tools/pub/pubspec#name), it should be `[a-z][a-z0-9_]*`. Lower-case only, not starting with a digit. (Can technically start with a `_`, but probably shouldn't.)

[Dart package names](https://github.com/dart-lang/language/blob/main/accepted/2.8/language-versioning/package-config-file-v2.md#package-names) are more permissive than Pub (to allow fx internal `foo.bar.bax` package names).
Decide which one this should match.

Reply:
(Just a drive-by - these are not regexes, see https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

Reply:
Right, `.` is not a special character here

Reply:
Oh great, not confusing at all! :grin: